File: ios/event/GeofenceEvent.m
//
//  GeofenceEvent.m
//  RNBackgroundLocation
//
//  Geofence Event
//  Android GeofenceEvent.java benzeri
//

#import "GeofenceEvent.h"
#import "LocationModel.h"

@implementation GeofenceEvent

- (instancetype)initWithIdentifier:(NSString *)identifier 
                             action:(GeofenceAction)action 
                           location:(LocationModel *)location {
    self = [super init];
    if (self) {
        _identifier = identifier;
        _action = action;
        _location = location;
    }
    return self;
}

- (NSDictionary *)toDictionary {
    NSMutableDictionary *dict = [NSMutableDictionary dictionary];
    dict[@"identifier"] = self.identifier;
    dict[@"action"] = [self actionString];
    dict[@"location"] = [self.location toDictionary];
    return dict;
}

- (NSString *)getEventName {
    return @"geofence";
}

- (NSString *)actionString {
    switch (self.action) {
        case GeofenceActionEnter:
            return @"ENTER";
        case GeofenceActionExit:
            return @"EXIT";
        case GeofenceActionDwell:
            return @"DWELL";
        default:
            return @"UNKNOWN";
    }
}

@end

File: android/src/main/java/com/backgroundlocation/event/GeofenceEvent.java
package com.backgroundlocation.event;

import org.json.JSONObject;

/**
 * Geofence Event
 * Geofence ENTER/EXIT/DWELL olayı
 */
public class GeofenceEvent {
    private final String identifier;
    private final String action; // ENTER, EXIT, DWELL
    private final JSONObject location;
    
    public GeofenceEvent(String identifier, String action, JSONObject location) {
        this.identifier = identifier;
        this.action = action;
        this.location = location;
    }
    
    public String getIdentifier() {
        return identifier;
    }
    
    public String getAction() {
        return action;
    }
    
    public JSONObject getLocation() {
        return location;
    }
    
    public JSONObject toJson() {
        try {
            JSONObject json = new JSONObject();
            json.put("identifier", identifier);
            json.put("action", action);
            json.put("location", location);
            return json;
        } catch (Exception e) {
            return new JSONObject();
        }
    }
    
    public String getEventName() {
        return "geofence";
    }
}



